test/Microsoft.ML.Tests/DatabaseLoaderTests.cs

```diff
@@ -0,0 +1,544 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Data;
+using System.Data.Common;
+using Microsoft.ML.Data;
+using Microsoft.ML.RunTests;
+using Microsoft.ML.TestFramework;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.ML.Tests
+{
+    public class DatabaseLoaderTests : BaseTestClass
+    {
+        public DatabaseLoaderTests(ITestOutputHelper output)
+            : base(output)
+        {
+        }
+
+        [Fact]
+        public void TryItOut()
+        {
+            var mlContext = new MLContext(seed: 1);
+
+            var loaderColumns = new DatabaseLoader.Column[]
+            {
+                new DatabaseLoader.Column() { Name = "Label", Type = DbType.Int32 },
+                new DatabaseLoader.Column() { Name = "SepalLength", Type = DbType.Single },
+                new DatabaseLoader.Column() { Name = "SepalWidth", Type = DbType.Single },
+                new DatabaseLoader.Column() { Name = "PetalLength", Type = DbType.Single },
+                new DatabaseLoader.Column() { Name = "PetalWidth", Type = DbType.Single }
+            };
+
+            using (var connection = new MockConnection(mlContext, GetDataPath(TestDatasets.iris.trainFilename), loaderColumns))
+            {
+                connection.Open();
+
+                using (var command = new MockCommand(connection, "SepalLength", "SepalWidth", "PetalLength", "PetalWidth"))
```

Comment:
You need to read the Label column, right? That's how the trainer is going to know what the expected value should be.

Reply:
The `MockCommand` doesn't actually do anything with these column names right now, I should remove them and/or update the mock to use them.

Reply:
Updated to have the `MockDbDataReader` filter based on the input column names.